awscrt/http.py

```diff
@@ -144,6 +145,84 @@ def on_incoming_headers_received(headers, response_code, has_body):
 
         return request
 
+
+'''
+TODO: Server Connection class,and configure the server connection.
+'''
+
+
+class HttpServer(object):
+    """
+    Represents an Http server. Everything in this class is non-blocking.
+    """
+    __slots__ = ('_bootstrap', '_tls_connection_options', '_on_incoming_connection',
+                 '_on_destroy_complete', '_native_handle', 'destroy_complete')
+
+    # don't call me, I'm private
+    def __init__(self, bootstrap, on_incoming_connection, on_destroy_complete, tls_connection_options):
+        assert isinstance(bootstrap, ServerBootstrap)
```

Comment:
Why not use the default constructor? I don't see the advantage of forcing users to call the static `HttpServer.new_server(...)` vs normal `HttpServer(...)` python construction